include/h2o.h

```diff
@@ -1155,6 +1159,10 @@ struct st_h2o_req_t {
          * abs-path of the request (unmodified)
          */
         h2o_iovec_t path;
+        /**
+         * protocol ('connect-udp' for RFC-based datagram support)
+         */
+        h2o_iovec_t protocol;
```

Comment:
I'm not sure if we want to have a field added here.

We already have `h2o_req_t::upgrade` that contains the [upgrade token](https://www.iana.org/assignments/http-upgrade-tokens/http-upgrade-tokens.xhtml) associated to the request.

Our HTTP/1 handler already uses this `h2o_req_t::upgrade`.

`:protocol` pseudo header field is simply the way of transferring the upgrade token in HTTP/2 and HTTP/3, and therefore the value should be stored into `h2o_req_t::upgrade` as well.

Doing so would streamline the handling of masque requests in the connect handler, as they can rely on this HTTP-version agnostic field.